src/System.CodeDom/tests/System/CodeDom/Compiler/CodeDomProviderTests.cs

```diff
@@ -134,6 +136,20 @@ public void CreateProvider_NoSuchLanguage_ThrowsConfigurationErrorsException(str
             AssertIsConfigurationErrorsException(ex2);
         }
 
+        [Fact]
+        public void CreateProvider_SerializeConfigurationErrorsException_ThrowsPlatformNotSupportedException()
+        {
+            Exception ex = Assert.ThrowsAny<Exception>(() => CodeDomProvider.CreateProvider(string.Empty));
+            AssertIsConfigurationErrorsException(ex);
+
+            // Because CodeDomProvider.CreateProvider(string.Empty) throws an internal ConfigurationErrorsException,
+            // rather than the publically exposed exception, we need to call BinaryFormatterHelpers.AssertExceptionDeserializationFails
+            // through reflection.
+            MethodInfo method = typeof(BinaryFormatterHelpers).GetMethod(nameof(BinaryFormatterHelpers.AssertExceptionDeserializationFails));
```

Comment:
Can we instead this add new overload to BinaryFormatterHelpers which takes Type to avoid reflection? (current one can call it since it's doing `typeof(T)` anyway)

Reply:
We’d have to do the same reflection jazz in that method in BinaryFormatter. I actually don’t think it would be really that useful. Mainly because this is the only case IIRC where we catch and validate an internal exception thrown from public API